tests/test_python_mono_repo.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from pathlib import Path
+import pytest
+
+from synthtool.languages import python_mono_repo
+from . import util
+
+
+TEST_FIXTURE_DIR = Path(__file__).parent / "fixtures" / "python_mono_repo"
+
+
+def test_apply_workaround():
+    with util.copied_fixtures_dir(TEST_FIXTURE_DIR):
+        docs_index_rst = Path(
+            "packages/google-cloud-workflows/docs/index.rst"
+        ).read_text()
+        assert (
+            docs_index_rst
+            == "First replacement\nFirst replacement\nSecond replacement\n"
+        )
+        python_mono_repo.apply_client_specific_post_processing(
+            "packages/google-cloud-workflows/scripts/client-post-processing",
+            "google-cloud-workflows",
+        )
+        docs_index_rst = Path(
+            "packages/google-cloud-workflows/docs/index.rst"
+        ).read_text()
+        assert (
+            docs_index_rst
+            == "Completed replacement 1.\nCompleted replacement 1.\nCompleted replacement 2.\n"
+        )
+
+        # Confirm that `AssertionError`` is raised if you run the post processing more than once
+        # because the replacement should only occur once.
+        with pytest.raises(AssertionError):
```

Comment:
I thought we said the script should be idempotent? I don't think this should raise.

(This issue is separate from but very related to the idempotency issue in the other thread.)

Reply:
I think the script is idempotent because running the script more than once won't cause unintended replacements. You would get the same generated code, regardless of how many times you run the script.

To add more background, the client specific post processing runs selectively only when the generator runs (when the directory `owl-bot-staging/<package-name>` exists). This logic is already part of the python post processor here:

https://github.com/googleapis/synthtool/blob/d52e638b37b091054c869bfa6f5a9fedaba9e0dd/synthtool/languages/python_mono_repo.py#L145

The current behaviour is:

- There is a copy-code job which is part of the owlbot design that copies code from googleapis-gen and puts it into `owl-bot-staging/<package name>`
- We only run the post processor logic when the `owl-bot-staging/<package name exists>`
- The client specific post processing will only during the post processing logic. 

This assert is checking that if we try to run the client specific post processing again without an `owl-bot-staging` folder, there should be no changes to ensure that if the client specific post processing again there won't be unintended replacements.

Reply:
I added a comment in f92efba to clarify that the method `apply_client_specific_post_processing` will only be called when `owl-bot-staging` is present.